plugin/storage/es/samplingstore/storage_test.go

```diff
@@ -0,0 +1,486 @@
+				Logger:          zap.NewNop(),
+				IndexPrefix:     test.prefix,
+				IndexDateLayout: "2006-01-02",
+				MaxDocCount:     defaultMaxDocCount,
+			})
+			assert.Equal(t, test.expected+samplingIndex, r.samplingIndexPrefix)
+		})
+	}
+}
+
+func TestGetReadIndices(t *testing.T) {
+	tests := []struct {
+		name  string
+		start time.Time
+		end   time.Time
+	}{
+		{
+			name:  "",
+			start: time.Date(2024, time.February, 10, 0, 0, 0, 0, time.UTC),
+			end:   time.Date(2024, time.February, 12, 0, 0, 0, 0, time.UTC),
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			expectedIndices := []string{
+				"prefix-jaeger-sampling-2024-02-12",
+				"prefix-jaeger-sampling-2024-02-11",
+				"prefix-jaeger-sampling-2024-02-10",
+			}
+			rollover := -time.Hour * 24
+			indices := getReadIndices("prefix-jaeger-sampling-", "2006-01-02", test.start, test.end, rollover)
+			assert.Equal(t, expectedIndices, indices)
+		})
+	}
+}
+
+func TestGetLatestIndices(t *testing.T) {
+	tests := []struct {
+		name            string
+		indexDateLayout string
+		maxDuration     time.Duration
+		expectedIndices []string
+		expectedError   string
+		IndexExistError error
+		indexExist      bool
+	}{
+		{
+			name:            "with index",
+			indexDateLayout: "2006-01-02",
+			maxDuration:     24 * time.Hour,
+			expectedIndices: []string{indexWithDate("", "2006-01-02", time.Now().UTC())},
+			expectedError:   "",
+			indexExist:      true,
+		},
+		{
+			name:            "without index",
+			indexDateLayout: "2006-01-02",
+			maxDuration:     72 * time.Hour,
+			expectedError:   "falied to find latest index",
+			indexExist:      false,
+		},
+		{
+			name:            "check index existence",
+			indexDateLayout: "2006-01-02",
+			maxDuration:     24 * time.Hour,
+			expectedError:   "failed to check index existence: fail",
+			indexExist:      true,
+			IndexExistError: errors.New("fail"),
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			withEsSampling("", test.indexDateLayout, defaultMaxDocCount, func(w *samplingStorageTest) {
+				indexService := &mocks.IndicesExistsService{}
+				w.client.On("IndexExists", mock.Anything).Return(indexService)
+				indexService.On("Do", mock.Anything).Return(test.indexExist, test.IndexExistError)
+				clientFnMock := w.storage.client()
+				actualIndices, err := getLatestIndices("", test.indexDateLayout, clientFnMock, -24*time.Hour, test.maxDuration)
+				if test.expectedError != "" {
+					require.EqualError(t, err, test.expectedError)
+					assert.Nil(t, actualIndices)
+				} else {
+					require.NoError(t, err)
+					require.Equal(t, test.expectedIndices, actualIndices)
+				}
+			})
+		})
+	}
+}
+
+func TestInsertThroughput(t *testing.T) {
+	tests := []struct {
+		name          string
+		expectedError string
+		esVersion     uint
+	}{
+		{
+			name:          "es v6",
+			expectedError: "",
+			esVersion:     6,
```

Comment:
did you mean to add v7 / v8? Is the logic actually different by version and against a mock storage? If not, there's no need to have a loop